consumer/consumererror/networkerrors.go

```diff
@@ -0,0 +1,89 @@
+// Copyright The OpenTelemetry Authors
+// SPDX-License-Identifier: Apache-2.0
+
+package consumererror // import "go.opentelemetry.io/collector/consumer/consumererror"
+
+import (
+	"fmt"
+	"net/http"
+
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/status"
+
+	"go.opentelemetry.io/collector/consumer/consumererror/internal/statusconversion"
+)
+
+// NetworkError contains either an HTTP or gRPC status code
```

Comment:
I think Network is very low level and may expect event errors from "net" package. Maybe "Transport"? Or as suggested we can have GRPCError and HTTPError as 2 structs.

Reply:
I'll do `Transport` for now, I see your point. See my other comment about a per-transport error type. I'm fine with those names if we go with two types.